src/services/user/services/QrRegistrationLinks.js

```diff
@@ -13,40 +13,97 @@ class QrRegistrationLinks {
 	}
 
 	async create(data) {
-		let { userIds } = data;
 		try {
-			userIds = await this.getUsersIdsWithoutAccount(userIds);
-			if (!userIds.length) {
+			const validUserIds = await this.getValidUserIds(data);
+			const userWithoutAccount = await this.getUsersIdsWithoutAccount(validUserIds);
+			if (!userWithoutAccount.length) {
 				return [];
 			}
-			return this.generateQrRegistrationLinks(userIds);
+			const generateServiceLinkParams = this.getGenerateLinkServiceParams(data);
+			const result = await this.generateQrRegistrationLinks(userWithoutAccount, generateServiceLinkParams);
+			return result;
 		} catch (err) {
 			throw new BadRequest(this.err.failed, err);
 		}
 	}
-	// TODO: remove awaits inside the loop
-	async generateQrRegistrationLinks(userIds) {
+
+	getGenerateLinkServiceParams(data) {
+		// save, patchUser. host,
+		const { save = true, patchUser = true, host } = data;
+		return { save, patchUser, host };
+	}
+
+	async getUserDetails(userIdChunk) {
+		return this.userModelService.find({
+			query: {
+				_id: userIdChunk,
+			},
+		});
+	}
+
+	async getHashLinks(userDetailList, generateServiceLinkParams) {
+		const promiseList = [];
+		for (const user of userDetailList) {
+			promiseList.push(
+				this.registrationLinkService
+					.create({
+						...generateServiceLinkParams,
+						role: user.roles[0],
+						schoolId: user.schoolId,
+						toHash: user.email,
+						hash: user.importHash,
+					})
+					.then(({ shortLink, hash }) => {
+						return {
+							hash,
+							email: user.email,
+							qrContent: shortLink,
+							firstName: user.firstName,
+							lastName: user.lastName,
+							title: `${user.firstName} ${user.lastName}`,
+							description: 'Zum Registrieren bitte den Link öffnen.',
+						};
+					})
+			);
+		}
+		const result = await Promise.all(promiseList);
+		return result;
+	}
+
+	async getUserHashes(userIdChunk, generateServiceLinkParams) {
+		const userDetailList = await this.getUserDetails(userIdChunk);
+		return this.getHashLinks(userDetailList.data, generateServiceLinkParams);
+	}
+
+	async generateQrRegistrationLinks(userIds, generateServiceLinkParams) {
 		const qrRegistrationLinks = [];
-		for (const userId of userIds) {
-			const user = await this.getUserInfo(userId);
-			const { shortLink } = await this.registrationLinkService.create({
-				role: user.roles[0],
-				save: true,
-				patchUser: true,
-				schoolId: user.schoolId,
-				toHash: user.email,
-			});
-			qrRegistrationLinks.push({
-				qrContent: shortLink,
-				title: `${user.firstName} ${user.lastName}`,
-				description: 'Zum Registrieren bitte den Link öffnen.',
-			});
+		let index = 0;
+		const arrayLength = userIds.length;
```

Comment:
you can inline for me this variable as something.length strictly indicated that it is indeed length
for (let index = 0; index < userIds.length

chunkSize can stay since it is more descriptiv than just number 200